### PR TITLE
[Issues:35] applied a coalesce to unit_source_value and unit_concept_id if exist

### PR DESCRIPTION
**Pull request by ChaoPang**

### CODE REVIEW

**Comment by ChaoPang** [comment]
@EthanSteinberg please take a look when you get a chance, thanks!

File: src/meds_etl/omop.py [hunk 1]
```diff
@@ -280,8 +280,11 @@ def write_event_data(
         if "visit_occurrence_id" in schema.names():
             metadata["visit_id"] = pl.col("visit_occurrence_id")
 
-        if "unit_source_value" in schema.names():
-            metadata["unit"] = pl.col("unit_source_value")
+        unit_columns = [pl.col(_) for _ in ["unit_source_value", "unit_concept_id"] if _ in schema.names()]
+        if unit_columns:
+            unit_expr = pl.coalesce(unit_columns)
+            unit_label_expr = unit_expr.replace_strict(concept_id_map, return_dtype=pl.Utf8(), default=None)
```
Comment (EthanSteinberg) [inline src/meds_etl/omop.py:286]: I don't think we want to do this replace if it's a unit_source_value?

I think we want:

```python
unit_columns = []
if "unit_source_value" in schema.names():
    unit_columns.append(pl.col("unit_source_value"))
if "unit_concept_id" in schema.names():
    unit_columns.append(pl.col("unit_concept_id").replace_strict(concept_id_map, return_dtype=pl.Utf8(), default=None))
  
if unit_columns:
      metadata["unit"] = pl.coalesce([unit_label_expr, unit_expr])
  
```

**Review by EthanSteinberg — CHANGES_REQUESTED** [review]
This looks mostly good, with one minor thing to fix.

Also, we will eventually want to write out a unit_source_value -> unit_concept_id map in the metadata, but no need to do that now

File: src/meds_etl/omop.py (hunk 1 above)
Comment (ChaoPang) [reply src/meds_etl/omop.py:286]: Good idea, will update

**Review by EthanSteinberg — APPROVED** [review]
Looks good thanks! Feel free to merge this whenever you are ready.

**Comment by ChaoPang** [comment]
@EthanSteinberg I don't think I will have permission to merge the PR.

**Comment by EthanSteinberg** [comment]
@ChaoPang interesting. I'll get you added to this repo. I just merged it in the meantime. Thanks!

**Pull request merged**